File: lab-3-ddb/environment/ruby_2.6.0/solution/edit_item.rb
require "aws-sdk-dynamodb"

dynamodb_client = Aws::DynamoDB::Client.new(region: "us-east-1")

table_name = "lostcats"
petname = "Hosepipe"

params = {
        table_name: table_name,
        key: {
                petname: petname
        },
        update_expression: "set breed = :b",
        expression_attribute_values: {
                ":b" => "British Shorthair"
        },
        return_values: "UPDATED_NEW"
}

begin
        dynamodb_client.update_item(params)
    puts "Updated Item"
end

File: lab-3-ddb/environment/node_8.10.0/edit_item.js
var
    AWS = require("aws-sdk"),
    DDB = new AWS.<FMI>({
        apiVersion: "2012-08-10",
        region: "<FMI>"
    });
(function editItemInDynamo(){
    var params = {
        <FMI>: "lostcats",
        Key:{
            petname: {
                S: "Hosepipe"
            }
        },
        UpdateExpression: "<FMI>",
        <FMI>: {
            ":b": {
                "S": "British Shorthair"
            }
        },
        ReturnValues: "UPDATED_NEW"
    };
    DDB.<FMI>(params, function(err, data){
        console.log(err, data);
    });
})();
